server/zoom/client.go

```diff
@@ -4,128 +4,24 @@
 package zoom
 
 import (
-	"bytes"
 	"encoding/json"
-	"net/http"
-	"net/url"
 
-	jwt "github.com/dgrijalva/jwt-go"
 	"github.com/mattermost/mattermost-server/v5/model"
-	"github.com/pkg/errors"
 )
 
-const (
-	zoomAPIKey     = "api.zoom.us"
-	zoomAPIVersion = "v2"
-	jwlAlgorithm   = "HS256"
-)
-
-type ClientError struct {
-	StatusCode int
-	Err        error
-}
-
-func (ce *ClientError) Error() string {
-	return ce.Err.Error()
-}
-
-// Client represents a Zoom API client
-type Client struct {
-	apiKey     string
-	apiSecret  string
-	httpClient *http.Client
-	baseURL    string
+// Client defines a common interface for the API and OAuth Zoom clients
+type Client interface {
+	GetMeeting(meetingID int) (*Meeting, error)
+	GetUser(user *model.User) (*User, *AuthError)
 }
```

Comment:
In go it's Idiomatik to define an `interface` where it's used. Should this interface be moved to `main` instead?